internal/validations/clusterlogforwarder/validate_collector_compatibility.go

```diff
@@ -0,0 +1,72 @@
+package clusterlogforwarder
+
+import (
+	v1 "github.com/openshift/cluster-logging-operator/api/logging/v1"
+	"github.com/openshift/cluster-logging-operator/internal/constants"
+	"github.com/openshift/cluster-logging-operator/internal/validations/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	vector = map[string]struct{}{
+		v1.OutputTypeElasticsearch:      {},
+		v1.OutputTypeGoogleCloudLogging: {},
+		v1.OutputTypeHttp:               {},
+		v1.OutputTypeKafka:              {},
+		v1.OutputTypeLoki:               {},
+		v1.OutputTypeSplunk:             {},
+		v1.OutputTypeSyslog:             {},
+		v1.OutputTypeCloudwatch:         {},
+		v1.OutputTypeAzureMonitor:       {},
+	}
+
+	fluentd = map[string]struct{}{
+		v1.OutputTypeElasticsearch:  {},
+		v1.OutputTypeFluentdForward: {},
+		v1.OutputTypeHttp:           {},
+		v1.OutputTypeKafka:          {},
+		v1.OutputTypeLoki:           {},
+		v1.OutputTypeSyslog:         {},
+		v1.OutputTypeCloudwatch:     {},
+	}
+)
+
+//validateCollectorCompatibility checking is given collector support proposed output if not error will return, nil otherwise
+//
+//Output type         | Supported collector type
+//Elasticsearch       | Fluentd, Vector
+//Fluent Forward      | Fluentd
+//Google Cloud Logging| Vector
+//HTTP                | Fluentd, Vector
+//Kafka               | Fluentd, Vector
+//Loki                | Fluentd, Vector
+//Splunk              | Vector
+//Syslog              | Fluentd, Vector
+//Amazon CloudWatch   | Fluentd, Vector
+//Azure Monitor Logs  | Vector
+func validateCollectorCompatibility(clf v1.ClusterLogForwarder, k8sClient client.Client, extras map[string]bool) (error, *v1.ClusterLogForwarderStatus) {
+	var collector string
```

Comment:
Consider just initializing these to what is supported for vector and swapping them if the collector is fluentd

Reply:
done